firestoreconv/firestoreconv.go

```diff
@@ -0,0 +1,35 @@
+package firestoreconv
+
+import (
+	"errors"
+	"time"
+
+	"github.com/googleapis/google-cloudevents-go/cloud/firestore/v1"
+
+	"github.com/benchttp/worker/internal"
+)
+
+// Package googleapis/google-cloud-go offers no native way of
+// converting the Firestore event format to a typed struct.
+// See https://github.com/googleapis/google-cloud-go/issues/1438
+// for more details.
+
+// ProtobufToType converts a Firestore event payload to a usable internal.Report.
+func ProtobufToType(v firestore.Value) (internal.Report, error) {
+	r := internal.Report{}
+
+	recordspb, ok := v.Fields["records"]
+	if !ok {
+		return internal.Report{}, errors.New("field does not exist") // TODO
+	}
+
+	for _, v := range recordspb.ArrayValue.Values {
+		timepb, ok := v.MapValue.Fields["time"]
+		if !ok {
+			return internal.Report{}, errors.New("")
```

Comment:
I guess this error is TO DO too?
We could write something like "field time does not exist" for the moment?